spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteManifestsAction.java

```diff
@@ -466,6 +474,309 @@ public void testRewriteLargeManifestsPartitionedTable() throws IOException {
+                            Conversions.fromByteBuffer(Types.IntegerType.get(), p.lowerBound()),
+                        (Integer)
+                            Conversions.fromByteBuffer(Types.IntegerType.get(), p.upperBound())))
+            .collect(Collectors.toList());
+
+    List<Integer> lowers = c3Boundaries.stream().map(t -> t.first()).collect(Collectors.toList());
+    List<Integer> uppers = c3Boundaries.stream().map(t -> t.second()).collect(Collectors.toList());
+
+    // The custom function sorts datafiles by having c3 bucket partitions in the range 0-4 and 5-9.
+    // Internally, that looks like
+    // +--------------------+---------------------+
+    // |partition           |__clustering_column__|
+    // +--------------------+---------------------+
+    // |{0, -531806488, 0}  |cluster=LT_5         |
+    // |{1, 385955472, 7}   |cluster=GTE_5        |
+    // |{2, 604077840, 6}   |cluster=GTE_5        |
+    // |{3, 1875302972, 4}  |cluster=LT_5         |
+    // |{4, -1772544904, 0} |cluster=LT_5         |
+    // |{5, 172551248, 7}   |cluster=GTE_5        |
+    // ...
+    // After rewriting, this looks like
+    // - manifest 1 -> [lower bound = 0, upper bound = 4]
+    // - manifest 2 -> [lower bound = 5, upper bound = 9]
+    // Without the custom sorting, each manifest tracks the full range of c3 upper/lower bounds.
+    // AKA they look like
+    // - manifest 1 -> [lower bound = 0, upper bound = 9]
+    // - manifest 2 -> [lower bound = 0, upper bound = 9]
+    // So the upper bound of the partitions tracked by the first file should be LEQ the lower bounds
+    // of the second. Etc
+    assertThat(uppers.get(0))
+        .withFailMessage(
+            "Upper bound of first manifest partition should be LESS THAN the lower bound of second")
+        .isLessThan(lowers.get(1));
+
+    // Each file should contain less than the full c3 partition span
+    c3Boundaries.forEach(
+        boundary -> {
+          assertThat(boundary.second() - boundary.first())
+              .withFailMessage(
+                  "Manifest should contain less than the full range of c3 bucket partitions")
+              .isLessThan(c3PartitionMax - c3PartitionMin);
+        });
+
+    // c3's Bucket(10) partition means our true lower bound = 0 and true upper bound is 9. The first
+    // manifest should include the lower bound of 0, and the last should have the upper bound of 9
+    assertThat(lowers.get(0))
+        .withFailMessage("Lower bound of first manifest c3 bucket partition should be 0")
+        .isEqualTo(0);
+    assertThat(uppers.get(0))
+        .withFailMessage("Upper bound of first manifest c3 bucket partition should be 4")
+        .isEqualTo(4);
+    assertThat(lowers.get(1))
+        .withFailMessage("Lower bound of second manifest c3 bucket partition should be 5")
+        .isEqualTo(5);
+    assertThat(uppers.get(1))
+        .withFailMessage("Upper bound of second manifest c3 bucket partition should be 9")
```

Comment:
all of these should use `.as()` instead of `.withFailMessage()`

Reply:
Ack